syntax = "proto3";

package bb8.apps;

service ContentInfo {
  rpc PersonalRecommend (RecommendRequest) returns (EntriesList) {}
  rpc Trending (TrendingRequest) returns (EntriesList) {}
  rpc Search (SearchRequest) returns (EntriesList) {}
  rpc GetContent (GetContentRequest) returns (EntryContent) {}
}

message Entry {
  string title = 1;
  string description = 2;
  string link = 3;
  string source = 4;
  string image_url = 5;
}

message EntriesList {
  repeated Entry entries = 1;
}

message SearchRequest {
  string query = 1;
  int32 user_id = 2;
  int32 count = 3;
}

message TrendingRequest {
  int32 user_id = 1;
  string source_name = 2;
  int32 count = 3;
}

message RecommendRequest {
  int32 user_id = 1;
  int32 count = 2;
}

message GetContentRequest {
  string entry_link = 1;
  int32 char_offset = 2;
  int32 limit = 3;
}

message EntryContent {
  string content = 1;
  int32 char_offset = 2;
}
